pkg/flowaggregator/collector/grpc.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2025 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+	"k8s.io/klog/v2"
+
+	flowpb "antrea.io/antrea/pkg/apis/flow/v1alpha1"
+)
+
+const grpcCollectorAddress = "0.0.0.0:14739"
+
+type grpcCollector struct {
+	service *grpcService
+	server  *grpc.Server
+}
+
+func NewGRPCCollector(recordCh chan *flowpb.Flow, caCert, serverKey, serverCert []byte) (*grpcCollector, error) {
+	cas := x509.NewCertPool()
+	if ok := cas.AppendCertsFromPEM(caCert); !ok {
+		return nil, fmt.Errorf("error when adding generate CA cert to pool")
+	}
+	cert, err := tls.X509KeyPair(serverCert, serverKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse key pair: %w", err)
+	}
+	tlsConfig := &tls.Config{
+		ClientAuth:   tls.RequireAndVerifyClientCert,
+		ClientCAs:    cas,
+		MinVersion:   tls.VersionTLS12,
+		Certificates: []tls.Certificate{cert},
+	}
+	service := &grpcService{
+		recordCh: recordCh,
+	}
+	server := grpc.NewServer(grpc.Creds(credentials.NewTLS(tlsConfig)))
+	flowpb.RegisterFlowExportServiceServer(server, service)
+	return &grpcCollector{
+		service: service,
+		server:  server,
+	}, nil
+}
+
+func (c *grpcCollector) Run(stopCh <-chan struct{}) {
+	// #nosec G102: binding to all network interfaces is intentional
+	lis, err := net.Listen("tcp", grpcCollectorAddress)
+	if err != nil {
+		klog.ErrorS(err, "Failed to listen on address", "addr", grpcCollectorAddress)
+		return
+	}
+	// c.server.Stop() will close the listener
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		klog.InfoS("Starting gRPC collector", "addr", lis.Addr().String())
+		if err := c.server.Serve(lis); err != nil {
```

Comment:
```suggestion
		if err := c.server.Serve(lis); err != nil && err != grpc.ErrServerStopped {
```
I suppose there is no need to log error when it's stopped?